File: N1085.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.Math;

public class N1085 {
	public static void main(String[] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		String input = br.readLine();
		int x = Integer.parseInt(input.split(" ")[0]);
		int y = Integer.parseInt(input.split(" ")[1]);
		int w = Integer.parseInt(input.split(" ")[2]);
		int h = Integer.parseInt(input.split(" ")[3]);
		
		System.out.print(Math.min(Math.min(x, y), Math.min(w-x, h-y)));
	}
}

File: N1085.js
var fs = require('fs');
var input = fs.readFileSync('/dev/stdin').toString().trim();
//var input = "6 2 10 3"
var data = new Array()

function setting() {
	var dots = input.split(' ')
	dots.forEach((e) => {
		data.push(parseInt(e))
	})
}

function solution() {
	console.log(Math.min(data[0], data[1], data[2]-data[0], data[3]-data[1]))
}

setting()
solution()
